middleware/assert_test.go

```diff
@@ -221,25 +325,20 @@ func TestAssertDatasetType(t *testing.T){
 				f := assert.DatasetType(next)
 				f.ServeHTTP(w, r)
 
-				Convey("The response should have body, status and headers as returned by dp-filter-flex-api", func(){
+				Convey("The response should have body, status and headers as returned by dp-filter-flex-api", func() {
 					So(len(datasetAPIMock.GetCalls()), ShouldEqual, 1)
 					So(len(filterFlexAPIMock.ForwardRequestCalls()), ShouldEqual, 1)
-					So(w.Code, ShouldEqual, expectedResponse.StatusCode)
+					So(w.Code, ShouldEqual, http.StatusOK)
 					So(w.HeaderMap.Get("X-Test"), ShouldResemble, "Value")
 					So(w.HeaderMap.Get("X-Foo"), ShouldResemble, "")
 					So(w.Body.String(), ShouldResemble, testBody)
 				})
 			})
 
-			Convey("When an incoming request passes through a disabled assert.FilterType middleware", func(){
-				assert := middleware.NewAssert(
-					responder.New(),
-					datasetAPIMock,
-					filterFlexAPIMock,
-					&mock.DataStore{},
-					testToken,
-					false,
-				)
+			Convey("When an incoming request passes through a disabled assert.FilterType middleware", func() {
+				assertIsEnabled := false
+				datasetIsFlexible := true
```

Comment:
These shadow vars in the same scope. I would probably just move the others into the block they're needed, or maybe just inline the bools at the callsite?

Reply:
Addressed.